azure-pipelines.yml

```diff
@@ -15,6 +15,17 @@ pr:
 variables:
 - template: ${{ variables.Pipeline.Workspace }}/.azure-pipelines/templates/variables.yml
 
+parameters:
+- name: architectures
+  type: object
+  default:
+  - name: AMD64
+    value: amd64
+    pool: pool-ubuntu-2004
```

Comment:
Can we reuse the variable defined in `.azure-pipelines/templates/variables.yml`? This can certainly reduce maintenance effort.

Reply:
No, because I need to define an object, variables can only define string to string mapping.

Reply:
I can't use variable in parameters.
`pool: ${{ variables.ubuntu_pool }}` leads to `/azure-pipelines.yml (Line: 24, Col: 11): A template expression is not allowed in this context`